diff --git a/_yaml.pxd b/_yaml.pxd
index 7937c9db51..d556de9217 100644
--- a/_yaml.pxd
+++ b/_yaml.pxd
@@ -9,7 +9,7 @@ cdef extern from "_yaml.h":
     char *PyString_AS_STRING(object o)
     int PyString_GET_SIZE(object o)
     object PyString_FromStringAndSize(char *v, int l)
-    object PyUnicode_FromString(char *u)
+    object PyUnicode_FromString(const char *u)
     object PyUnicode_DecodeUTF8(char *u, int s, char *e)
     object PyUnicode_AsUTF8String(object o)
     int PY_MAJOR_VERSION
diff --git a/_yaml.pyx b/_yaml.pyx
index ff4efe80b5..05eed472ef 100644
--- a/_yaml.pyx
+++ b/_yaml.pyx
@@ -2,7 +2,7 @@
 import yaml
 
 def get_version_string():
-    cdef char *value
+    cdef const char *value
     value = yaml_get_version_string()
     if PY_MAJOR_VERSION < 3:
         return value
