diff --git b/__init__.py a/__init__.py
index aa329fb..19c2ffe 100644
--- b/__init__.py
+++ a/__init__.py
@@ -1,3 +1,3 @@
-from .core import where, old_where
+from .core import where, old_where, ca_cert_data
 
 __version__ = "2018.08.24"
diff --git b/core.py a/core.py
index eab9d1d..7753eef 100644
--- b/core.py
+++ a/core.py
@@ -33,5 +33,14 @@ def old_where():
     )
     return where()
 
+
+_ca_cert_data = None
+def ca_cert_data():
+    global _ca_cert_data
+    if _ca_cert_data is None:
+        _ca_cert_data = __loader__.get_data(where()).decode("utf-8")
+    return _ca_cert_data
+
+
 if __name__ == '__main__':
     print(where())
