diff --git a/core.py b/core.py
index 91f538bb1f..c89fadf417 100644
--- a/core.py
+++ b/core.py
@@ -46,7 +46,7 @@ if sys.version_info >= (3, 11):
     def contents() -> str:
         return files("certifi").joinpath("cacert.pem").read_text(encoding="ascii")
 
-elif sys.version_info >= (3, 7):
+elif False:  # sys.version_info >= (3, 7):
 
     from importlib.resources import path as get_path, read_text
 
@@ -90,6 +90,8 @@ else:
     Package = Union[types.ModuleType, str]
     Resource = Union[str, "os.PathLike"]
 
+    _ca_cert_data = None
+
     # This fallback will work for Python versions prior to 3.7 that lack the
     # importlib.resources module but relies on the existing `where` function
     # so won't address issues with environments like PyOxidizer that don't set
@@ -100,8 +102,10 @@ else:
         encoding: str = 'utf-8',
         errors: str = 'strict'
     ) -> str:
-        with open(where(), encoding=encoding) as data:
-            return data.read()
+        global _ca_cert_data
+        if _ca_cert_data is None:
+            _ca_cert_data = __loader__.get_data(where()).decode(encoding)
+        return _ca_cert_data
 
     # If we don't have importlib.resources, then we will just do the old logic
     # of assuming we're on the filesystem and munge the path directly.
