diff --git a/client/_code_cache.py b/client/_code_cache.py
index 2212805d69..31d6166660 100644
--- a/client/_code_cache.py
+++ b/client/_code_cache.py
@@ -21,7 +21,7 @@ def _ensure_list(path):
 
 
 def _find_gen_dir():
-    """Create, if needed, and return a directory where automatically
+    r"""Create, if needed, and return a directory where automatically
     generated modules will be created.
 
     Usually, this is the directory 'Lib/site-packages/comtypes/gen'.
