diff --git a/client/_code_cache.py b/client/_code_cache.py
index 2212805d69..31d6166660 100644
--- a/client/_code_cache.py
+++ b/client/_code_cache.py
@@ -20,7 +20,7 @@ def _ensure_list(path):
 
 
 def _find_gen_dir():
-    """Create, if needed, and return a directory where automatically
+    r"""Create, if needed, and return a directory where automatically
     generated modules will be created.
 
     Usually, this is the directory 'Lib/site-packages/comtypes/gen'.
diff --git a/client/_generate.py b/client/_generate.py
index 1e94e1e0a6..4f5bb71851 100644
--- a/client/_generate.py
+++ b/client/_generate.py
@@ -29,7 +29,7 @@ def _name_module(tlib):
     return "comtypes.gen." + modname
 
 def GetModule(tlib):
-    """Create a module wrapping a COM typelibrary on demand.
+    r"""Create a module wrapping a COM typelibrary on demand.
 
     'tlib' must be an ITypeLib COM pointer instance, the pathname of a
     type library, a COM CLSID GUID, or a tuple/list specifying the
