diff --git b/adapters.py a/adapters.py
index fa4d9b3..6b80628 100644
--- b/adapters.py
+++ a/adapters.py
@@ -30,7 +30,8 @@ from urllib3.exceptions import LocationValueError
 
 from .models import Response
 from .compat import urlparse, basestring
-from .utils import (DEFAULT_CA_BUNDLE_PATH, extract_zipped_paths,
+# Add DEFAULT_CA_BUNDLE_DATA.
+from .utils import (DEFAULT_CA_BUNDLE_PATH, DEFAULT_CA_BUNDLE_DATA, extract_zipped_paths,
                     get_encoding_from_headers, prepend_scheme_if_needed,
                     get_auth_from_url, urldefragauth, select_proxy)
 from .structures import CaseInsensitiveDict
@@ -224,12 +225,18 @@ class HTTPAdapter(BaseAdapter):
                 cert_loc = extract_zipped_paths(DEFAULT_CA_BUNDLE_PATH)
 
             if not cert_loc or not os.path.exists(cert_loc):
-                raise IOError("Could not find a suitable TLS CA certificate bundle, "
-                              "invalid path: {}".format(cert_loc))
+                # Check DEFAULT_CA_BUNDLE_DATA.
+                if not DEFAULT_CA_BUNDLE_DATA:
+                    raise IOError("Could not find a suitable TLS CA certificate bundle, "
+                                  "invalid path: {}".format(cert_loc))
 
             conn.cert_reqs = 'CERT_REQUIRED'
 
-            if not os.path.isdir(cert_loc):
+            # Set ca_cert_data if cert_loc is not specified.
+            # if not os.path.isdir(cert_loc):
+            if not cert_loc or not os.path.exists(cert_loc):
+                conn.ca_cert_data = DEFAULT_CA_BUNDLE_DATA
+            elif not os.path.isdir(cert_loc):
                 conn.ca_certs = cert_loc
             else:
                 conn.ca_cert_dir = cert_loc
diff --git b/certs.py a/certs.py
index d1a378d..7d63b3b 100644
--- b/certs.py
+++ a/certs.py
@@ -13,6 +13,11 @@ environment, you can change the definition of where() to return a separately
 packaged CA bundle.
 """
 from certifi import where
+# Import ca_cert_data for single binarisation.
+try:
+    from certifi import ca_cert_data
+except ImportError:
+    pass
 
 if __name__ == '__main__':
     print(where())
diff --git b/utils.py a/utils.py
index 0ce7fe1..3260d1f 100644
--- b/utils.py
+++ a/utils.py
@@ -37,6 +37,11 @@ from .exceptions import (
 NETRC_FILES = ('.netrc', '_netrc')
 
 DEFAULT_CA_BUNDLE_PATH = certs.where()
+# Load ca_cert_data for single binarisation.
+if hasattr(certs, "ca_cert_data"):
+    DEFAULT_CA_BUNDLE_DATA = certs.ca_cert_data()
+else:
+    DEFAULT_CA_BUNDLE_DATA = None
 
 
 if sys.platform == 'win32':
