diff --git a/connectionpool.py b/connectionpool.py
index 4708c5bfc7..b514ebf055 100644
--- a/connectionpool.py
+++ b/connectionpool.py
@@ -900,6 +900,7 @@ class HTTPSConnectionPool(HTTPConnectionPool):
         assert_hostname=None,
         assert_fingerprint=None,
         ca_cert_dir=None,
+        ca_cert_data=None,
         **conn_kw
     ):
 
@@ -924,6 +925,7 @@ class HTTPSConnectionPool(HTTPConnectionPool):
         self.key_password = key_password
         self.ca_certs = ca_certs
         self.ca_cert_dir = ca_cert_dir
+        self.ca_cert_data = ca_cert_data
         self.ssl_version = ssl_version
         self.assert_hostname = assert_hostname
         self.assert_fingerprint = assert_fingerprint
@@ -942,6 +944,7 @@ class HTTPSConnectionPool(HTTPConnectionPool):
                 cert_reqs=self.cert_reqs,
                 ca_certs=self.ca_certs,
                 ca_cert_dir=self.ca_cert_dir,
+                ca_cert_data=self.ca_cert_data,
                 assert_hostname=self.assert_hostname,
                 assert_fingerprint=self.assert_fingerprint,
             )
